### PR TITLE
Rename `isHud` to `respectCamera`

### PR DESCRIPTION
**Pull request by spydon**
# Description

This PR also renames `isHud` to `respectViewport`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame/lib/src/components/component.dart [hunk 1]
```diff
@@ -130,8 +131,10 @@ class Component with Loadable {
 
   void render(Canvas canvas) {}
 
-  void renderTree(Canvas canvas) {
-    render(canvas);
+  void renderTree(Canvas canvas, {bool callRender = true}) {
```
Comment (erickzanardo) [inline packages/flame/lib/src/components/component.dart:134]: What if we call callRender to skipOwnRender, to make it clear that the flag is about to the component own render and not it's children?

File: packages/flame/lib/src/game/camera/camera_wrapper.dart [hunk 2]
```diff
@@ -1,34 +1,38 @@
...
 import 'camera.dart';
 
 /// This class encapsulates FlameGame's rendering functionality. It will be
 /// converted into a proper Component in a future release, but until then
 /// using it in any code other than the FlameGame class is unsafe and
 /// not recommended.
-class CameraWrapper {
-  // TODO(st-pasha): extend from Component
+@internal
+class CameraWrapper extends Component {
```
Comment (erickzanardo) [inline packages/flame/lib/src/game/camera/camera_wrapper.dart:13]:
```suggestion
class _CameraWrapper extends Component {
```

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/lib/src/game/camera/camera_wrapper.dart (hunk 2 above)
Comment (spydon) [reply packages/flame/lib/src/game/camera/camera_wrapper.dart]: If it is private we have to define it inside of `flame_game.dart` right? `@internal` and not being exported should be enough?

File: packages/flame/lib/src/components/component.dart [hunk 3]
```diff
@@ -18,11 +18,12 @@ import 'cache/value_cache.dart';
...
-  /// Whether this component is a HUD (Heads-up display) object or not.
+  /// Whether this component should respect the camera or not.
   ///
-  /// HUD objects ignore the FlameGame.camera when rendered (so their position
-  /// coordinates are considered relative to the device screen).
-  bool isHud = false;
+  /// Components that have this property set to false will ignore the
+  /// FlameGame.camera when rendered (so their position coordinates are
+  /// considered relative only to the viewport instead).
+  bool respectCamera = true;
```
Comment (st-pasha) [inline packages/flame/lib/src/components/component.dart:26]: Maybe add a note here that the property is provisional, and we plan to remove it post-v1?

File: packages/flame/lib/src/game/flame_game.dart [hunk 4]
```diff
@@ -17,17 +17,15 @@ import 'projector.dart';
 /// This is the recommended base class to use for most games made with Flame.
 /// It is based on the Flame Component System (also known as FCS).
 class FlameGame extends Component with Game {
+  late final CameraWrapper _cameraWrapper;
```
Comment (st-pasha) [inline packages/flame/lib/src/game/flame_game.dart:20]: Note that Flutter style guide recommends that constructors would go first in the class, before any variables are declared: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#constructors-come-first-in-a-class.
There is even a lint rule for that, though we haven't enabled it yet.

**Review by luanpotter — APPROVED** [review]
LGTM

File: packages/flame/lib/src/components/component.dart (hunk 3 above)
Comment (spydon) [reply packages/flame/lib/src/components/component.dart:26]: I don't think that is necessary yet, we don't even have a plan for how to remove it?

File: packages/flame/lib/src/game/flame_game.dart (hunk 4 above)
Comment (spydon) [reply packages/flame/lib/src/game/flame_game.dart:20]: TIL, it's different in pretty much all other OOP languages that I've programmed in. Will change when I get home!

File: packages/flame/lib/src/components/component.dart (hunk 3 above)
Comment (luanpotter) [reply packages/flame/lib/src/components/component.dart:26]: I would just say that it is only respected for top-level components, and we can think of whether we want to change that interface or not later

File: packages/flame/lib/src/game/flame_game.dart (hunk 4 above)
Comment (luanpotter) [reply packages/flame/lib/src/game/flame_game.dart:20]: That is very unusual to me too, I think we can decide whether we want that rule or not separately as I imagine there are many, many violations on the entire project

**Pull request merged**